addons/web/static/src/core/position_hook.js

```diff
@@ -118,12 +118,12 @@ function getBestPosition(reference, popper, { container, margin, position }) {
     };
     /** @type {VariantsData} */
     const variantsData = {
-        vs: refBox.left,
+        vs: refBox.left + popMargins.left,
         vm: refBox.left + refBox.width / 2 + -popBox.width / 2,
-        ve: refBox.right - popBox.width,
-        hs: refBox.top,
+        ve: refBox.right - popMargins.right - popBox.width,
+        hs: refBox.top + popMargins.top,
         hm: refBox.top + refBox.height / 2 + -popBox.height / 2,
-        he: refBox.bottom - popBox.height,
+        he: refBox.bottom - popMargins.bottom - popBox.height,
```

Comment:
Considering the fact that all of these values either apply on the position:
`left` (vertically placed popper), or `top` (horizontally placed popper)
my reasoning would be: we want to ignore the impact of the style margin on the popper in the "variant" position, thus the margins that are to be considered are always `margin-left` and `margin-top`, resulting in the following `variantsData`:
```js
    const variantsData = {
        vs: refBox.left - popMargins.left,
        vm: refBox.left + refBox.width / 2 + -popBox.width / 2 - popMargins.left,
        ve: refBox.right - popMargins.left - popBox.width,
        hs: refBox.top - popMargins.top,
        hm: refBox.top + refBox.height / 2 + -popBox.height / 2 - popMargins.top,
        he: refBox.bottom - popMargins.top - popBox.height,
    };
```